server/src/main/java/com/broadcom/lsp/cobol/service/CobolLanguageServer.java

```diff
@@ -124,14 +127,53 @@ private void getLocaleFromClient() {
     settingsService.getConfiguration(LOCALE.label).thenAccept(localeStore.notifyLocaleStore());
   }
 
+  @Override
+  public int getExitCode() {
+    return exitCode;
+  }
+
   @Override
   public CompletableFuture<Object> shutdown() {
-    return supplyAsync(() -> TRUE);
+    LOG.info("LS for COBOL received shutdown request");
+    exitCode = 0;
+    try {
+      cancelAllProcessing();
+    } catch (Exception exception) {
+      return supplyAsync(() -> new ShutdownResponse(null, exception.getMessage()));
```

Comment:
You can use `CompletableFuture.completedFuture` in both returns.

Reply:
done